contracts/Trader.sol

```diff
@@ -67,7 +67,7 @@ contract Trader is Ownable {
         require(n > 0, "TDR: Received empty arrays");
 
         // gas cost for individual traders
-        uint256 gasCost = (tx.gasprice * tx.gasused) / (2 * n);
+        uint256 gasCost = (tx.gasprice * gasleft()) / (2 * n);
```

Comment:
@sporejack this would mean we are subsidising all gas up to this point. Should this be `gasused + gasleft()`?